### PR TITLE
Update food after a change to the values, not on focusout

### PR DESCRIPTION
**Pull request by marshallhouston**
#### Pivotal URL:
https://www.pivotaltracker.com/story/show/154822219

#### What does this PR do?
This PR adds an event listener to save the original values of the food and then passes that original value into the updateFood function. If the current values are different than the original value, it sends a fetch request to the database. Otherwise, a fetch request is not sent.

#### Where should the reviewer start?
eventListeners.js to see the setOriginalFoodListener that creates the event listener in index.js and grabs the original values.

#### Any additional context you want to provide?
We added this card because a fetch request was being sent every time that we tabbed through foods.html, even if the values hadn't changed.

I've also defined an empty object, originalFood, that gets passed into the eventListeners file, and it then gets passed into the foodsRequests.updateFoods method. This raised a code smell to me, but I couldn't come up with another way to make it accessible.  

#### Screenshots (if appropriate)
Gif of the network tab. The only time it send an actual request is once the values have changed.

updateonchange

### CODE REVIEW

File: lib/requests/foods.js [hunk 1]
```diff
@@ -79,21 +79,23 @@ const renderFood = (food) => {
...
   var foodCalories = $(`.food-item-${id}`).children()[1].innerText
-  fetch(`https://ivmh-qs-api.herokuapp.com/api/v1/foods/${id}`, {
-    method: 'PUT',
-    headers: {'Content-Type': 'application/json'},
-    body: JSON.stringify({
-      food: {
-        name: foodName,
-        calories: foodCalories
-      }
+  if (originalFood.name != foodName || originalFood.calories != foodCalories ) {
```
Comment (izaxveta) [inline lib/requests/foods.js]: Perhaps we could pull out the functionality where the original value is checked against the value when focused out into another function - and only call updateFood when the food name/calorie value has actually changed.

**Comment by marshallhouston** [comment]
@izaxveta I moved the check of original value with the current value out of the updateFood method. It is only called if the value has changed.

File: lib/index.js [hunk 2]
```diff
@@ -32,7 +32,11 @@ $(document).ready(() => {
     var fileName = location.pathname.split('/').slice(-1)[0]
     if (fileName === 'foods.html' && event.target.nodeName != 'BUTTON') {
       var foodId = event.target.parentElement.attributes.data.value.split('-')[1]
```
Comment (izaxveta) [inline lib/index.js:34]: We could probably also line 34-36 into another function that simply sets these variables, and then also pull out the if statement into another that simply checks for any change in original value upon focus out.

Comment (izaxveta) [reply lib/index.js:34]: I think this works for now and we can continue to pull out on later iterations after the completion of the express backend transition.

**Pull request merged**